lib/CodeGen/CodeGenModule.cpp

```diff
@@ -3294,6 +3302,10 @@ void CodeGenModule::EmitGlobalFunctionDefinition(GlobalDecl GD,
                                                    /*DontDefer=*/true,
                                                    ForDefinition));
 
+  if (D->getAttr<CXXAMPRestrictAMPAttr>()) {
+    cast<llvm::Function>(GV)->addFnAttr("HC");
```

Comment:
This is the main reason why in the PR for LLVM we only check for "HC" function attribute.